dkg-gadget/src/signing_manager/mod.rs

```diff
@@ -203,84 +206,182 @@ where
-				let concat_data = dkg_pub_key
-					.clone()
-					.into_iter()
-					.chain(unsigned_proposal_bytes.clone())
-					.chain(ssid.encode())
-					.collect::<Vec<u8>>();
-				let seed = sp_core::keccak_256(&concat_data);
-
-				let maybe_set = self
-					.generate_signers(&seed, threshold, best_authorities.clone(), dkg_worker)
-					.ok();
-				if let Some(signing_set) = maybe_set {
-					// if we are in the set, send to work manager
-					if signing_set.contains(&party_i) {
-						dkg_worker.logger.info(format!(
-							"🕸️  Session Id {:?} | SSID {} | {}-out-of-{} signers: ({:?})",
-							session_id,
-							ssid,
-							threshold,
-							best_authorities.len(),
-							signing_set,
-						));
-
-						let params = SigningProtocolSetupParameters::MpEcdsa {
-							best_authorities: best_authorities.clone(),
-							authority_public_key: authority_public_key.clone(),
-							party_i,
-							session_id,
-							threshold,
-							stage: ProtoStageType::Signing { unsigned_proposal_hash },
-							unsigned_proposal_batch: batch.clone(),
-							signing_set,
-							associated_block_id: *header.number(),
-							ssid,
-						};
-
-						let signing_protocol = dkg_worker
-							.dkg_modules
-							.get_signing_protocol(&params)
-							.expect("Standard signing protocol should exist");
-						match signing_protocol.initialize_signing_protocol(params).await {
-							Ok((handle, task)) => {
-								// Send task to the work manager. Force start if the type chain ID
-								// is None, implying this is a proposal needed for rotating sessions
-								// and thus a priority
-								let force_start = typed_chain_id == TypedChainId::None;
-								self.work_manager.push_task(
-									unsigned_proposal_hash,
-									force_start,
-									handle,
-									task,
-								)?;
-							},
-							Err(err) => {
-								dkg_worker
-									.logger
-									.error(format!("Error creating signing protocol: {:?}", &err));
-								dkg_worker.handle_dkg_error(err.clone()).await;
-								return Err(err)
-							},
+				self.maybe_add_signing_set(
+					dkg_worker,
+					&dkg_pub_key,
+					&unsigned_proposal_bytes,
+					ssid,
+					threshold,
+					session_id,
+					party_i,
+					&best_authorities,
+					&mut signing_sets,
+					false,
+				);
+
+				// If this is the last signing set, then, we should add any wildcard sets
+				// Definition of "wildcard set": a signing set that may or may not be symmetric
+				// to other generated signing sets since we exclude blamed part indexes from the
+				// generated signing set and replace with non-blamed parties. Since blames are not
+				// necessarily symmetric between nodes, it follows that these wildcard sets are
+				// not necessarily symmetric either. As such, the only way a wildcard set converges
+				// successfully to completion is if the same node(s) is blamed on all nodes
+				// participating in the signing protocol.
+				if ssid == MAX_POTENTIAL_SIGNING_SETS_PER_PROPOSAL - 1 {
```

Comment:
imo this blame logic will fire if any node is slow to respond too. Right now we have an elegant flow to enqueue those messages, but this blaming could make the situation worse, by starting t+1 asymmetric signing sets.